module/os/freebsd/zfs/zfs_vfsops.c

```diff
@@ -1835,6 +1835,12 @@ zfs_fhtovp(vfs_t *vfsp, fid_t *fidp, int flags, vnode_t **vpp)
 		return (SET_ERROR(EINVAL));
 	}
 
+	if (fidp->fid_len == LONG_FID_LEN && (fid_gen > 1 || setgen != 0)) {
+		dprintf("snapdir fid: fid_gen (%llu) and setgen (%llu)\n",
+		    fid_gen, setgen);
```

Comment:
I think we've been casting these to `u_longlong_t` for printing elsewhere in this file.

Reply:
Indeed; updated.